src/MAUI/Maui.Samples/Samples/Data/GeodatabaseTransactions/GeodatabaseTransactions.xaml.cs

```diff
@@ -41,22 +43,53 @@ public partial class GeodatabaseTransactions : ContentPage
         // Store a reference to the table to be edited.
         private GeodatabaseFeatureTable _editTable;
 
+        // Flag indicating if an edit exists which is not committed and was created during a transaction.
+        // Not to be confused with HasLocalEdits() which is for local edits not synchronized with the service.
+        private bool _hasUncommittedEdits;
+
         public GeodatabaseTransactions()
         {
             InitializeComponent();
 
-            // When the spatial reference changes (the map loads) add the local geodatabase tables as feature layers.
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // When the map view loads, add the local geodatabase tables as feature layers.
             MyMapView.SpatialReferenceChanged += async (s, e) =>
             {
-                // Call a function (and await it) to get the local geodatabase (or generate it from the feature service).
-                await GetLocalGeodatabase();
+                try
+                {
+                    // Call a function (and await it) to get the local geodatabase (or generate it from the feature service).
+                    await GetLocalGeodatabase();
 
-                // Once the local geodatabase is available, load the tables as layers to the map.
-                await LoadLocalGeodatabaseTables();
+                    // Once the local geodatabase is available, load the tables as layers to the map.
+                    await LoadLocalGeodatabaseTables();
+                }
+                catch (Exception ex)
+                {
+                    // Show the exception message.
+                    MessageTextBlock.Text = ex.Message;
+                }
+
+                // Create a graphic for the geodatabase extent.
+                var lineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, System.Drawing.Color.Red, 2);
+                var extentGraphic = new Graphic(_extent, lineSymbol);
+
+                // Create a graphics overlay for the extent graphic and apply a renderer.
+                var extentOverlay = new GraphicsOverlay
+                {
+                    Graphics = { extentGraphic },
+                    Renderer = new SimpleRenderer(lineSymbol)
+                };
+
+                // Add graphics overlay to the map view.
+                MyMapView.GraphicsOverlays.Add(extentOverlay);
```

Comment:
I think adding the GraphicsOverlay with SimpleRenderer do not need to wait for SR changed and can be done upfront.

What may need to wait is adding a graphic with the extent geometry. If this happens outside try-catch, is there guarantee/check `_extent`  is not null or empty?

Reply:
Good eye! Moving this out of the event handler. We can mark `_extent` as `readonly`.